### DISCUSSION TITLE
How to get the current value of a SpringValue

### DISCUSSION THREAD

**Discussion by LoganDark** [pmndrs/react-spring#1663] [Support]
## 🤓 Question

Hey, I'm having trouble getting the current value of a SpringValue. So far, this has worked well enough:

```ts
			const animationValues = value.animation.values.map(v => v.getValue())
			let currentValue

			// SpringValue.get() isn't reliable, it can return undefined randomly, so we have to poke into the internals
			if (animationValues.length === 0) {
				// this spring hasn't animated yet, i.e. you have a `to` but not a `from`. We can grab the `to` though,
				// to get the 'current' value
				currentValue = value.animation.to

			} else {
				// this spring is animating / has animated, it knows where it is, it has an animated value. grab it
				currentValue = animationValues[0]
			}
```

However, it needs a couple renders for the SpringValue to even be populated with its target, which means this doesn't exactly work. Here's a screenshot with the SpringValue before and after it has a target

![image](https://user-images.githubusercontent.com/4723091/129425921-5e4c2d72-7b38-448f-b98f-d2465c3b5543.png)

As you can see, the top one has no `to` - or to be more accurate the `to` is `undefined` - I have checked and it does in fact exist when `useSpring` is called - the SpringValue is just populated late. This is causing my application to break briefly as you can see by the error messages.

Is there any way to reliably get the value of a spring? Currently the `animated` HOC does not work for my purposes, especially since refs aren't supported

**Comment by joshuaellis** [comment]
Hi, It'd be great to understand your use case for this? Maybe a minimal code example would help too?

**Reply by LoganDark** [reply]
Replicating `animated` so that it works with ForwardRef'd components

**Reply by LoganDark** [reply]
It also really helps with things like unmounting a component when it's fully invisible (say, `visibility` spring value is 0), which is currently impossible without tricking `animated` with wrapper components

**Reply by gormonn** [reply]
@joshuaellis For example: https://github.com/pmndrs/react-spring/issues/109
to get opacity and then do display: none

**Reply by gormonn** [reply]
useChain was been helpful
